src/libstore/http-binary-cache-store.cc

```diff
@@ -115,7 +115,11 @@ class HttpBinaryCacheStore : public BinaryCacheStore
 
     FileTransferRequest makeRequest(const std::string & path)
     {
-        FileTransferRequest request(cacheUri + "/" + path);
+        auto uri = cacheUri + "/" + path;
+        if (hasPrefix(path, "https://") || hasPrefix(path, "http://") || hasPrefix(path, "file://")) {
+          auto uri = path;
+        }
```

Comment:
I've tried `if .. else` here but declaring `auto uri` in each block resulted in a compiler error.

Reply:
This doesn't do anything since `auto uri` is scoped to the if-block. Do you mean `uri = path`?